File: aug22_20/at2.py
if __name__ == "__main__":
    x = input()
    x = int(x)

    digits = 200000
    index = 200000

    tmp1 = []
    for i in range(digits):
        tmp2 = pow(2,index)

        if(tmp2>x):
            continue
        tmp3 = x%tmp2
        if tmp2 != 0:
            x -= tmp3*index
            tmp1.append(tmp3)
        index -= 1
    result1 = sum(tmp1)

    if(result1%9 ==0):
        print("Yes")
    else:
        print("No")

File: aug22_20/at2.cpp
#include <iostream>
#include <vector>
#include <math.h>
#include <cstdint>

using namespace std;


int main(){
    
    unsigned int N;
    N = 999999999999;
    // cin >> N;

    int digits = 200000;
    int div;

    vector<int> tmp1(digits);
    int tmp2;
    for(int i=digits; i>=0; i--){
        div = pow(10, i);
        tmp2 = ceil(N/div);

        if(div>N){
            continue;
        }
        

        if(tmp2 != 0){
            N -= div*tmp2;
            tmp1[i] = tmp2;
        }
        
        // cout << tmp2 << endl;

    }

    int result = 0;
    for(int i=0; i<digits; i++){
        result += tmp1[i];
    }

    int result2 = result%9;

    string ans;

    if(result2==0){
        ans="Yes";
    }else{
        ans = "No";
    }
    cout << ans << endl;
    // cout << result2 << endl;
}